### ISSUE TITLE
Add possibility to list available tests

### ISSUE THREAD

**Issue by tempora-mutantur** [pester/Pester#2278]
### Checklist

- [X] Feature request has a meaningful title
- [X] I have searched the existing issues. See [all issues](https://github.com/pester/Pester/issues?q=is%3Aissue)
- [X] I have tested using the latest version of Pester. See [Installation and update guide](https://pester.dev/docs/introduction/installation).

### Summary of the feature request

Hi!

We use Pester and we love it. We invested a lot in our tests and now they take more time than we want them to.

These tests are hard to run in parallel on one VM and we want to split them into several chunks to run in multiple VMs/containers.

We would like to have a Pester command that would list available tests. We could use this information to divide them into several chunks and run them in parallel. (similar to what is suggested in this article - https://learn.microsoft.com/en-us/azure/devops/pipelines/test/parallel-testing-any-test-runner?view=azure-devops).

Some other testing frameworks have such possibility, like NUnit that has the "--explore" option. https://docs.nunit.org/articles/nunit/running-tests/Console-Command-Line.html

It would simplify getting information about available tests in Pester. This information could be used as an input for ConfigurationProperty Filter.FullName to run only those tests that you need.

Thank you!

### How should it work?

It might be a separate cmdlet or a switch/configuration property. It launches pester, runs the discovery and instead of running tests it saves the information about the tests in a file.

**Comment by tempora-mutantur** [comment]
That exactly what I need. Thank you and sorry for disturbing.

Maybe I can extend the documentation so it's easier to find these options?

**Comment by fflaten** [comment]
No disturbance, glad I could help! 🙂 

Currently it's only documented as part of [`New-PesterConfiguration`](https://pester.dev/docs/commands/New-PesterConfiguration). Feel free to submit a PR in [docs-repo](https://github.com/pester/docs) if you have a suitable place in mind. I don't have a suggestion personally atm since it's more of a advanced use case.

**Comment by tempora-mutantur** [comment]
@fflaten 

Hi. Sorry, I have one more question about Pester's filters and options to select specific tests

I have a PesterConfiguration with two filter options set - $configuration.Filter.Tag and $configuration.Filter.FullName.

I expected, that the both filters will apply to the subset of tests and it'll narrow down the search so only specified tests that has the selected tag will be run. Pester works in a different manner. It creates a union of two filters so all the tests specified in $configuration.Filter.FullName will be launched as well as all the tests having $configuration.Filter.Tag tags. 

If that's the strategy we expect from Pester, maybe we could mention this behavior it in the New-PesterConfiguration Filter documentation.

**Comment by fflaten** [comment]
The union behavior is by-design. See #1442 for the original plan (focus-option is not available).

Yes, this should be better documented - good call!
I've created an issue in docs-repo to update https://pester.dev and another issue here to consider mentioning it in the category-description in the configuration-object, ex.
![image](https://user-images.githubusercontent.com/3436158/209587848-abde757b-a418-4454-a19b-c6c5a86a51e1.png)


_For the future: This is a separate issue/topic at minimum and probably also better suited for github discussions or discord. Just so we don't revive closed topics and can discuss it properly. 🙂  _
